sfm_pc/views.py

```diff
@@ -295,13 +298,24 @@ def osm_autocomplete(request):
           ST_Y(ST_Centroid(geometry)) AS latitude,
           ST_asgeojson(ST_Simplify(geometry, 0.01))::json AS geojson
         FROM osm_data
-        WHERE plainto_tsquery('english', %s) @@ search_index
     '''
 
+    if search_by_id:
+        # We want to search for text that begins with the query term, which uses
+        # SQL "string%" wildcard syntax. But because of the Django DB API's
+        # particular string formatting idiosyncracies, we need to append the %
+        # directly to the search term before formatting it into the query.
```

Comment:
Terrific - that makes sense....that's a bit annoying about Django, though.

Reply:
Yeah, it was an annoying half hour spent trying to figure out why it wasn't working. But in general the built-in protections against SQL injection in Django are really comforting!